### PR TITLE
Add an option to allow setting owner password on output

### PR DESCRIPTION
**Pull request by bhaugen-abledocs**
## Jira Issue Link(s)
- [ADAPI-309](https://axespdf.atlassian.net/browse/ADAPI-309) -- internal issue tracking system

## What does this PR change?
- adds an option to allow setting an owner password on the output file

## TODO
**N/A**

## Known Issues
**N/A**

## Requirements for Reviewer
- [ ] Are the parts of the code that are complex documented?
- [ ] Has any other supplemental documentation been updated?
- [ ] Does the code have automated tests?
- [ ] Does the code follow agreed upon code conventions?

### CODE REVIEW

File: watermark-utility/Watermarker.cs [hunk 1]
```diff
@@ -50,19 +51,30 @@ public static int Main(string[] args)
...
                 Console.WriteLine($"The value for --input is : {input}");
                 Console.WriteLine($"The value for --ouptput is : {output}");
                 Console.WriteLine($"The value for --pages is : {pages}");
                 Console.WriteLine($"The value for --text is : {text}");
                 Console.WriteLine($"The value for --image is : {image}");
 
+                if (outputPassword is null)
+                    Console.WriteLine($"The value for --outputPassword is : null");
+                else
+                    Console.WriteLine($"The value for --outputPassword is not null");
```
Comment (abledocs-mkuznicki) [inline watermark-utility/Watermarker.cs:76]: 👍

File: watermark-utility/Watermarker.cs [hunk 2]
```diff
@@ -92,7 +104,24 @@ public static int Main(string[] args)
...
+                // Create the PdfWriter that is going to be used
+                PdfWriter writer;
+                if (outputPassword is not null)
+                {
+                    // if a password was provider, use it as the Owner Password
+                    WriterProperties writerProperties = new WriterProperties();
+                    writerProperties.SetStandardEncryption(
+                        null, // no user password
+                        Encoding.UTF8.GetBytes(outputPassword), // set owner password
+                        EncryptionConstants.ALLOW_PRINTING,
```
Comment (abledocs-mkuznicki) [inline watermark-utility/Watermarker.cs:116]:
```suggestion
                        EncryptionConstants.ALLOW_PRINTING | EncryptionConstants.ALLOW_SCREENREADERS,
--


,
```

**Review by abledocs-mkuznicki — COMMENTED** [review]
Looks good! One change desired (made as suggestions) to explicitly allow screen readers, as PDF before 2.0 could restrict the ability to read-aloud which causes issues with many screen readers.

File: watermark-utility/Watermarker.cs (hunk 2 above)
Comment (bhaugen-abledocs) [reply watermark-utility/Watermarker.cs]: This has been applied and spacing fixed up.

**Comment by bhaugen-abledocs** [comment]
> Looks good! One change desired (made as suggestions) to explicitly allow screen readers, as PDF before 2.0 could restrict the ability to read-aloud which causes issues with many screen readers.

Ah, interesting! The suggestion has been applied.

File: watermark-utility/Watermarker.cs (hunk 2 above)
Comment (abledocs-mkuznicki) [reply watermark-utility/Watermarker.cs]: Thanks!

**Review by abledocs-mkuznicki — APPROVED** [review]

**Pull request merged**